community/values/src/main/java/org/neo4j/values/storable/TemporalValue.java

```diff
@@ -259,7 +262,7 @@ private TemporalValue attachTimeZone( TemporalValue temporal, ZoneId zoneIdToAtt
             }
             else
             {
-                throw new IllegalStateException( "Should only attach offsets to local times, not zone ids." );
+                throw new UnsupportedTemporalUnitException( "Should only attach offsets to local times, not zone ids." );
```

Comment:
Since we had an `IllegalStateException` before, is this code path a case we ever expect to happen? If not, an `IllegalStateException` is the right choice, since it signals that _we_ did wrong, not the user of the product.